clients/src/main/java/org/apache/kafka/common/requests/LeaveGroupResponse.java

```diff
@@ -55,6 +56,22 @@ public LeaveGroupResponse(LeaveGroupResponseData data) {
         this.data = data;
     }
 
+    public LeaveGroupResponse(LeaveGroupResponseData data, short version) {
+        super(ApiKeys.LEAVE_GROUP);
+
+        if (version >= 3) {
+            this.data = data;
+        } else {
+            if (data.members().size() != 1) {
+                throw new UnsupportedVersionException("LeaveGroup response version " + version +
+                    " can only contain one member, got " + data.members().size() + " members.");
+            }
+
+            this.data = new LeaveGroupResponseData()
```

Comment:
This seems to be normalizing the response? Seems like we just need to set the error code, but its done in a way that's a little confusing. Why do we need to getError when we have the data.errorCode?

Reply:
That's right. It is because it can be the top level error OR the member error. This is how it was done before, see L82.

Reply:
Are you referring to
`final short errorCode = getError(topLevelError, memberResponses).code();` ?
I guess it is the same, but done in more lines. The arguments to getError are also a bit clearer.

But maybe not a huge deal

Reply:
it might be more readable if we use a `topLevelError` variable as in L82, the line is a bit hard to understand